File: server/DB/models/Cart.model.js
import { Schema, model } from "mongoose";

const cartSchema = new Schema(
    {
        user: {
            type: Schema.Types.ObjectId,
            ref: "User",
            required: true,
        },

        items: [
            {
                product: {
                    type: Schema.Types.ObjectId,
                    ref: "Product",
                },
                quantity: Number,
            },
        ],

        totalPrice: {
            type: Number,
            required: true,
            default: 0
        }
    },
    { timestamps: true }
)

const cartModel = model("Cart", cartSchema)
export default cartModel

File: Restaurant/Core/Home/Models/CartModel.swift
//
//  CartModel.swift
//  Restaurant
//
//  Created by Mahmoud Nagdy on 23/01/2026.
//

import Foundation


struct CartModel: Codable, Identifiable {
    let id: String
    let items: [CartItem]
    let totalPrice: Double
    let user: String
    
    enum CodingKeys: String, CodingKey {
        case id = "_id", items, totalPrice, user
    }
}


struct CartItem: Codable, Identifiable{
    let id: String
    let product: ProductModel
    let quantity: Int
    
    enum CodingKeys: String, CodingKey {
        case id = "_id", product, quantity
    }
}

struct CartResponseModel: Codable{
    let message: String
    let cart: CartModel?
}

struct updateCartModel: Codable{
    let productId: String
    let quantity: Int
}
